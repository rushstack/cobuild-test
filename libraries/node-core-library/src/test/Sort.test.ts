// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

import { Sort } from '../Sort';

test('Sort.compareByValue', () => {
  const array: number[] = [3, 6, 2];
  array.sort(Sort.compareByValue); // [2, 3, 6]
});

test('Sort.compareByValue cases', () => {
  const values: unknown[] = [undefined, null, -1, 1];
  const results: string[] = [];
  for (let i: number = 0; i < values.length; ++i) {
    for (let j: number = 0; j < values.length; ++j) {
      const x: unknown = values[i];
      const y: unknown = values[j];
      let relation: string = '?';
      switch (Sort.compareByValue(x, y)) {
        case -1:
          relation = '<';
          break;
        case 0:
          relation = '=';
          break;
        case 1:
          relation = '>';
          break;
      }
      results.push(`${x} ${relation} ${y}`);
    }
  }
  expect(results).toMatchSnapshot();
});

test('Sort.sortBy', () => {
  const array: string[] = ['aaa', 'bb', 'c'];
  Sort.sortBy(array, (x) => x.length); // [ 'c', 'bb', 'aaa' ]
});

test('Sort.isSortedBy', () => {
  const array: string[] = ['a', 'bb', 'ccc'];
  Sort.isSortedBy(array, (x) => x.length); // true
});

test('Sort.sortMapKeys', () => {
  const map: Map<string, number> = new Map<string, number>();
  map.set('zebra', 1);
  map.set('goose', 2);
  map.set('aardvark', 3);
  Sort.sortMapKeys(map);
  expect(Array.from(map.keys())).toEqual(['aardvark', 'goose', 'zebra']);
});

test('Sort.sortSetBy', () => {
  const set: Set<string> = new Set<string>();
  set.add('aaa');
  set.add('bb');
  set.add('c');
  Sort.sortSetBy(set, (x) => x.length);
  expect(Array.from(set)).toEqual(['c', 'bb', 'aaa']);
});

test('Sort.sortSet', () => {
  const set: Set<string> = new Set<string>();
  set.add('zebra');
  set.add('goose');
  set.add('aardvark');
  Sort.sortSet(set);
  expect(Array.from(set)).toEqual(['aardvark', 'goose', 'zebra']);
});
