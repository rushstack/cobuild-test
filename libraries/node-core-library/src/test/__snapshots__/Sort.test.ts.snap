// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort.compareByValue cases 1`] = `
Array [
  "undefined = undefined",
  "undefined < null",
  "undefined < -1",
  "undefined < 1",
  "null > undefined",
  "null = null",
  "null < -1",
  "null < 1",
  "-1 > undefined",
  "-1 > null",
  "-1 = -1",
  "-1 < 1",
  "1 > undefined",
  "1 > null",
  "1 > -1",
  "1 = 1",
]
`;
